torch/_inductor/index_propagation.py

```diff
@@ -267,15 +287,37 @@ def inner(*args: Any, **kwargs: Any) -> IndexPropResult:
 
         return inner
 
+    def statically_true(self, e):
+        evaluated = self.shape_env._maybe_evaluate_static(
+            e,
+            axioms=self.axioms,
+            var_to_range=self.var_to_range,
+        )
+        return bool(evaluated)
```

Comment:
This seems like the wrong place for such reasoning. Would it make sense for `sizevars.statically_known_true` to inherit these axioms from somewhere?